python/hsml/core/dataset_api.py

```diff
@@ -19,52 +19,186 @@
 import json
 from hsml.client.exceptions import RestAPIError
 import time
+import copy
+from tqdm.auto import tqdm
 
 from hsml import client, tag
+from concurrent.futures import ThreadPoolExecutor, wait
 
+class Chunk:
+    def __init__(self, content, number, status):
+        self.content = content
+        self.number = number
+        self.status = status
+        self.retries = 0
 
 class DatasetApi:
     def __init__(self):
         pass
 
     DEFAULT_FLOW_CHUNK_SIZE = 1048576
+    FLOW_PERMANENT_ERRORS = [404, 413, 415, 500, 501]
 
-    def upload(self, local_abs_path, upload_path):
-        """Upload file/directory in local path to datasets
+    def upload(
+        self,
+        local_path: str,
+        upload_path: str,
+        overwrite: bool = False,
+        chunk_size=1048576,
+        simultaneous_uploads=3,
+        max_chunk_retries=1,
+        chunk_retry_interval=1,
+    ):
+        """Upload a file to the Hopsworks filesystem.
+
+        ```python
+
+        import hopsworks
```

Comment:
Same here, if we want to get rid of `hopsworks`. But I think it's fine to have (or encourage) the use of hopsworks library to connect.